README.md

```diff
@@ -1,2 +1,142 @@
+To add (or remove...) network interfaces from a running pod, the user should
+simply edit the running pod network selection elements - i.e. the `k8s.v1.cni.cncf.io/networks`
+annotation.
+
+Thus, if we had this running pod (and `NetworkAttachmentDefinition`):
+```yaml
+---
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: macvlan1-config
+spec:
+  config: '{
+            "cniVersion": "0.4.0",
+            "plugins": [
+                {
+                    "type": "macvlan",
+                    "capabilities": { "ips": true },
+                    "master": "eth1",
+                    "mode": "bridge",
+                    "ipam": {
+                        "type": "static"
+                    }
+                }, {
+                    "type": "tuning"
+                } ]
+        }'
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: macvlan1-worker1
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[
+            {
+                "name": "macvlan1-config",
+                "ips": [ "10.1.1.11/24" ]
+            }
+    ]'
+  labels:
+    app: macvlan
+spec:
+  containers:
+  - name: macvlan-worker1
+    image: docker.io/library/alpine:latest
+    command: ["/bin/sleep", "10000"]
+```
+
+We would run this example yaml to **add** an interface from it:
+```yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: macvlan1-worker1
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[
+            {
+                "name": "macvlan1-config",
+                "ips": [ "10.1.1.11/24" ]
+            },
+            {
+                "name": "macvlan1-config",
+                "interface": "ens4",
+            }
+    ]'
+  labels:
+    app: macvlan
+spec:
+  containers:
+    - name: macvlan-worker1
+      image: docker.io/library/alpine:latest
+      command: ["/bin/sleep", "10000"]
+```
+
+And we would run this example yaml to **remove** an interface to it:
+```yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: macvlan1-worker1
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[]' # this will remove all networks from the pod
+  labels:
+    app: macvlan
+spec:
+  containers:
+    - name: macvlan-worker1
+      image: docker.io/library/alpine:latest
+      command: ["/bin/sleep", "10000"]
+```
+
+## Configuration
+The `multus-dynamic-networks-controller` configuration is encoded in JSON, and allows the following keys:
+
+- `"criSocketPath"`: specify the path to the CRI socket. Defaults to `/run/containerd/containerd.sock`.
+- `"criType"`: either `crio` or `containerd`. Defaults to `containerd`.
+- `"multusSocketDir"`: directory where the multus socket will be found. Defaults to `/var/run/multus-cni/`.
+
+## Developer workflow
+Below you can find information on how to push local code changes to a kind cluster.
+
+- change code :)
+- start up a kind cluster. I've tested using the [multus repo e2e kind cluster](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/e2e/setup_cluster.sh).
```

Comment:
Is this kind cluster different from a vanilla one?

Reply:
It is not actually required - you can use whatever you want. 

... but everything will be connected in the underlying network. Without multiple interfaces in the host, that's not happening.